### DISCUSSION TITLE
VTI interfaces are not getting re-created in server, in a case where client IP is changed multiple times.

### DISCUSSION THREAD

**Discussion by ravimittal03** [strongswan/strongswan#606] [Q&A]
In a scenario, where two clients are connected to a server via IPsec tunnel and  client's ip is changed multiple times and there is already one or more client that have used the same IP earlier.
What exactly happens is, in the initial phase two clients (client 1 and client 2) are connected to a server and both have different mark value (suppose client 1= 3 and client2= 4), both the clients are assigned public IPs.
Now if the public IP of a client2 is changed such that the currently assigned IP was used by the client1, then it is observed in the logs that the mark value of client2 changes from 4 ->3.

Traffic got disrupted since vti4 peer public-IP and ipsec profile & tunnel IP does not match for mark value(mark=4 from swanctl -l). Hence Traffic and BGP tunnel were seen down.

post_ipsec_tunnel_notification comes from strongswan, here data sent from strongswan is wrong.

**Comment by tobiasbrunner** [comment]
Please see #196.

> post_ipsec_tunnel_notification

What's that?
